File: C++ Programming/Competitive Programming/Coding Challenges 13/program3.cpp
/*
3.Write a program which accept number from user and return the count of digits in
between 3 and 7.

Input : 2395
Output : 1

Input : 1018
Output : 0

Input : 4521
Output : 2

Input : 9922
Output : 0
*/

#include <iostream>
using namespace std;

int CountRange(int iNo)
{
    int iCount = 0;

    if (iNo < 0)
    {
        iNo = -iNo;
    }

    for (; iNo != 0; iNo = iNo / 10)
    {
        int iDigit = iNo % 10;

        if (iDigit == 4 || iDigit == 5 || iDigit == 6)
        {
            iCount++;
        }
    }

    return iCount;
}

int main()
{
    int iValue = 0;
    int iRet = 0;

    cout << "Enter number: ";
    cin >> iValue;

    iRet = CountRange(iValue);

    cout << "Count of digits between 3 and 7 is: " << iRet;

    return 0;
}

File: Java Programming/Competitive Programming/Coding Challenges 17/program3.java
/*
3.Write a program which accept number from user and count frequency of 2 in it.
Input : 2395
Output : 1
Input : 1018
Output : 0
Input : 9000
Output : 0
Input : 922432
Output : 3
*/

import java.util.Scanner;

class program3
{
    static int CountTwo(int iNo)
    {
        int iCount = 0;

        String str = Integer.toString(iNo);

        for (int i = 0; i < str.length(); i++)
        {
            if (str.charAt(i) == '2')
            {
                iCount++;
            }
        }

        return iCount;
    }

    public static void main(String[] args)
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0;
        int iRet = 0;

        System.out.print("Enter number: ");
        iValue = sobj.nextInt();

        iRet = CountTwo(iValue);

        System.out.print(iRet);

        sobj.close();
    }
}